django/db/migrations/operations/fields.py

```diff
@@ -299,49 +268,7 @@ def deconstruct(self):
         )
 
     def state_forwards(self, app_label, state):
-        model_state = state.models[app_label, self.model_name_lower]
-        # Rename the field
-        fields = model_state.fields
-        try:
-            found = fields.pop(self.old_name)
-        except KeyError:
-            raise FieldDoesNotExist(
-                "%s.%s has no field named '%s'" % (app_label, self.model_name, self.old_name)
-            )
-        fields[self.new_name] = found
-        for field in fields.values():
-            # Fix from_fields to refer to the new field.
-            from_fields = getattr(field, 'from_fields', None)
-            if from_fields:
-                field.from_fields = tuple([
-                    self.new_name if from_field_name == self.old_name else from_field_name
-                    for from_field_name in from_fields
-                ])
-        # Fix index/unique_together to refer to the new field
-        options = model_state.options
-        for option in ('index_together', 'unique_together'):
-            if option in options:
-                options[option] = [
-                    [self.new_name if n == self.old_name else n for n in together]
-                    for together in options[option]
-                ]
-        # Fix to_fields to refer to the new field.
-        delay = True
-        references = get_references(
-            state, (app_label, self.model_name_lower), (self.old_name, found),
-        )
-        for *_, field, reference in references:
-            delay = False
-            if reference.to:
-                remote_field, to_fields = reference.to
-                if getattr(remote_field, 'field_name', None) == self.old_name:
-                    remote_field.field_name = self.new_name
-                if to_fields:
-                    field.to_fields = tuple([
-                        self.new_name if to_field_name == self.old_name else to_field_name
-                        for to_field_name in to_fields
-                    ])
-        state.reload_model(app_label, self.model_name_lower, delay=delay)
+        state.rename_field(app_label, self.model_name, self.old_name, self.new_name)
```

Comment:
Looks like `self.model_name_lower` could be passed here and we could avoid all the `.lower()` handling in `.rename_field`.

```suggestion
        state.rename_field(app_label, self.model_name_lower, self.old_name, self.new_name)
```